### PR TITLE
Optional delta-mode

### PR DESCRIPTION
**Pull request by zvezdochiot**
1.6
* fix UB #14: optional delta-mode

### CODE REVIEW

File: xsct.1 [hunk 1]
```diff
@@ -12,6 +12,9 @@ sets the screen's color temperature.
 
 .SH OPTIONS
 .TP
+.B -d, --delta
+Xsct work in delta-mode
```
Comment (faf0) [inline xsct.1:16]:
```suggestion
Shift temperature by given argument
```

File: xsct.c [hunk 2]
```diff
@@ -184,8 +185,8 @@ static void sct_for_screen(Display *dpy, int screen, int temp, int fdebug)
 
 int main(int argc, char **argv)
 {
-    int i, screen, screens, temp;
-    int fdebug = 0, fhelp = 0;
+    int i, screen, screens, temp, tempd;
```
Comment (faf0) [inline xsct.c:188]:
```suggestion
    int i, screen, screens, temp;
```

`tempd` can be declared and initialized further down

File: xsct.c [hunk 3]
```diff
@@ -218,10 +220,32 @@ int main(int argc, char **argv)
...
+                for (screen = 0; screen < screens; screen++)
+                {
+                   sct_for_screen(dpy, screen, temp, fdebug);
+                }
+            }
+            else
+            {
+                for (screen = 0; screen < screens; screen++)
+                {
+                    tempd = temp + get_sct_for_screen(dpy, screen, fdebug);
```
Comment (faf0) [inline xsct.c:240]:
```suggestion
                    int tempd = temp + get_sct_for_screen(dpy, screen, fdebug);
```

File: xsct.c [hunk 4]
```diff
@@ -218,10 +220,32 @@ int main(int argc, char **argv)
         }
         else
         {
-            temp = (temp == 0) ? TEMPERATURE_NORM : temp;
-
-            for (screen = 0; screen < screens; screen++)
-                sct_for_screen(dpy, screen, temp, fdebug);
+            temp = ((temp == 0) && (fdelta == 0)) ? TEMPERATURE_NORM : temp;
+            if (fdelta == 0)
+            {
```
Comment (faf0) [inline xsct.c:225]:
```suggestion
            if (fdelta == 0)
            {
                if (temp == 0) {
                    temp = TEMPERATURE_NORM;
                }
```

File: xsct.c [hunk 5]
```diff
@@ -31,12 +31,13 @@
...
 static void usage(char * pname)
 {
-    printf("Xsct (1.5)\n"
+    printf("Xsct (1.6)\n"
            "Usage: %s [options] [temperature]\n"
            "\tIf the argument is 0, xsct resets the display to the default temperature (6500K)\n"
            "\tIf no arguments are passed, xsct estimates the current display temperature\n"
            "Options:\n"
            "\t-v, --verbose \t xsct will display debugging information\n"
+           "\t-d, --delta \t xsct work in delta-mode\n"
```
Comment (faf0) [inline xsct.c:40]:
```suggestion
           "\t-d, --delta \t xsct will shift temperature by given argument\n"
```

File: xsct.c [hunk 6]
```diff
@@ -218,10 +220,32 @@ int main(int argc, char **argv)
...
+                    fprintf(stderr, "WARNING! Temperatures below %d cannot be displayed.\n", TEMPERATURE_ZERO);
+                    temp = TEMPERATURE_ZERO;
+                }
+                for (screen = 0; screen < screens; screen++)
+                {
+                   sct_for_screen(dpy, screen, temp, fdebug);
+                }
+            }
+            else
+            {
```
Comment (faf0) [inline xsct.c:237]:
```suggestion
            {
                // Shift temperature by `temp` argument
```

**Comment by faf0** [comment]
@zvezdochiot do you want me to make a pull request based on this PR?

Or feel free to reopen this. You can commit the suggested changes directly over the GitHub web interface if you agree with the proposed changes.

File: xsct.c [hunk 7]
```diff
@@ -218,10 +220,32 @@ int main(int argc, char **argv)
...
         else
         {
-            temp = (temp == 0) ? TEMPERATURE_NORM : temp;
-
-            for (screen = 0; screen < screens; screen++)
-                sct_for_screen(dpy, screen, temp, fdebug);
+            temp = ((temp == 0) && (fdelta == 0)) ? TEMPERATURE_NORM : temp;
+            if (fdelta == 0)
+            {
+                if (temp < TEMPERATURE_ZERO)
```
Comment (faf0) [inline xsct.c:226]:
```c
            if (fdelta == 0)
            {
                if (temp == 0)
                {
                    temp = TEMPERATURE_NORM;
                }
                else if (temp < TEMPERATURE_ZERO)
```

Comment (zvezdochiot) [reply xsct.c:226]: Hi @faf0 .

Not. I want you to make your own using mine as a template. Too many fixes.

Comment (faf0) [reply xsct.c:226]: implemented here: https://github.com/faf0/sct/pull/18

**Pull request closed without merging**